src/test/java/uk/gov/ida/eventemitter/EventEmitterAmazonEventSenderTest.java

```diff
@@ -0,0 +1,124 @@
+                        new BasicAWSCredentials("", ""),
+                        Regions.EU_WEST_2);
+
+        final ExpectedRequest expectedRequest = new ExpectedRequest(AUDIT_EVENTS_API_RESOURCE, "POST", null, null);
+
+        try {
+            apiGatewayStub.register(expectedRequest, expectedResponse);
+            Response<Void> response = amazonEventSender.sendAuthenticated(aTestEventMessage().build(), "encryptedEvent");
+            Fail.fail("Should return Forbidden (403)");
+        } catch (AwsResponseException e) {
+            assertThat(e.getResponse().getStatusCode()).isEqualTo(403);
+        }
+    }
+
+    @Test
+    public void shouldReturnHTTP200WhenAuthenticationSucceeds() throws Exception {
+
+        final RegisteredResponse expectedResponse = new RegisteredResponse(
+                200,
+                "application/json",
+                "hello",
+                createTestResponseHeadersMap()
+        );
+        final EventSender amazonEventSender =
+                new AmazonEventSender(URI.create(apiGatewayStub.baseUri().build().toString() + AUDIT_EVENTS_API_RESOURCE),
+                        new BasicAWSCredentials(DUMMY_AWS_ACCESS_KEY, DUMMY_AWS_SECRET_ACCESS_KEY),
+                        Regions.EU_WEST_2);
+
+        /*
+            Expected headers not provided as HttpStub requires both header names and content to match, and content is dynamic.
+            The expected headers are tested separated using assertions in the try block below.
+         */
+        final ExpectedRequest expectedRequest =
+                new ExpectedRequest(AUDIT_EVENTS_API_RESOURCE, "POST", null, null);
+
+
+        apiGatewayStub.register(expectedRequest, expectedResponse);
+        Response<Void> response = amazonEventSender.sendAuthenticated(aTestEventMessage().build(), "encryptedEvent");
+        assertThat(response.getHttpResponse().getStatusCode()).isEqualTo(200);
+
+            /*
+                Check that all the headers required for AWS Authentication and AWS4 Signing are present.
+                Header contents are not validated.
+             */
+        assertThat(apiGatewayStub.getLastRequest().getHeader("Host")).isNotNull();
+        assertThat(apiGatewayStub.getLastRequest().getHeader("Authorization")).isNotNull();
+
+        // Authorization header must contain all of the following keywords...
+        assertThat(apiGatewayStub.getLastRequest().getHeader("Authorization")).containsPattern(Pattern.compile("(?=.*AWS4-HMAC-SHA256)(?=.*Signature)(?=.*Credential)"));
+        assertThat(apiGatewayStub.getLastRequest().getHeader("Content-type")).isEqualToIgnoringCase("application/json");
+        assertThat(apiGatewayStub.getLastRequest().getHeader("X-Amz-Date")).isNotNull();
+
+    }
+
+    @Test(expected = AwsResponseException.class)
+    public void shouldThrowAwsResponseExceptioWhenNotFound() throws java.io.UnsupportedEncodingException {
```

Comment:
Can we replace `java.io.UnsupportedEncodingException` with `UnsupportedEncodingException`?